File: api/go/tenki/hash/hash.go
package main

import (
	"crypto/md5"
	"encoding/binary"
	"flag"
	"fmt"
)

func toInt(date, secret string) int {
	str := fmt.Sprintf("%s %s", date, secret)
	sum := md5.Sum([]byte(str))
	num := binary.BigEndian.Uint32(sum[:4])
	return int(num)
}

func main() {
	var date string
	var secret string
	flag.StringVar(&date, "d", "", "date string")
	flag.StringVar(&secret, "s", "happyhalloween", "secret")
	flag.Parse()

	// Tue, 27 Oct 2015 08:46:40 JST ==> 51498849
	// Tue, 27 Oct 2015 08:46:41 JST ==> 1366538060
	// Tue, 27 Oct 2015 08:46:42 JST ==> 1443194715
	n := toInt(date, secret)
	fmt.Println(n)
}

File: api/go/tenki/hash/hash.py
# -*- coding: utf-8 -*-

"""
$ python3 hash.py --date="Tue, 27 Oct 2015 08:46:40 JST"
51498849
"""

import argparse
import hashlib
import struct

def toint(date, secret):
	s = date + " " + secret
	m = hashlib.md5(s.encode("utf-8"))
	d = m.digest()
	n = struct.unpack(">I", d[0:4])[0]
	return n

parser = argparse.ArgumentParser()
parser.add_argument('--date', type=str)
parser.add_argument('--secret', type=str, default="happyhalloween")
args = parser.parse_args()

n = toint(args.date, args.secret)
print("%s ==> %d" % (args.date, n))
